plenum/test/bls/test_sign_validation_for_key_proof_exist.py

```diff
@@ -0,0 +1,95 @@
+    for n in txnPoolNodeSet:
+        monkeypatch.setattr(n.poolManager.reqHandler, 'doStaticValidation', lambda req: True)
+        n.bls_bft.bls_key_register._pool_manager.config.VALIDATE_SIGN_WITHOUT_BLS_KEY_PROOF = True
+    new_blspk = update_bls_keys(0, sdk_wallet_stewards, sdk_pool_handle, looper, txnPoolNodeSet)
+    monkeypatch.undo()
+    check_bls_key(new_blspk, txnPoolNodeSet[0], txnPoolNodeSet)
+
+
+def test_switched_on_sign_validation_for_key_proof_exist(looper,
+                                                         txnPoolNodeSet,
+                                                         sdk_pool_handle,
+                                                         sdk_wallet_stewards,
+                                                         sdk_wallet_client,
+                                                         monkeypatch):
+    '''
+    Add BLS key without BLS key proof. Test that when VALIDATE_SIGN_WITHOUT_BLS_KEY_PROOF = False
+    node does not use key sent without proof.
+    '''
+
+    for n in txnPoolNodeSet:
+        monkeypatch.setattr(n.poolManager.reqHandler, 'doStaticValidation', lambda req: True)
+        n.bls_bft.bls_key_register._pool_manager.config.VALIDATE_SIGN_WITHOUT_BLS_KEY_PROOF = False
+    update_bls_keys(0, sdk_wallet_stewards, sdk_pool_handle, looper, txnPoolNodeSet)
+    monkeypatch.undo()
+    for n in txnPoolNodeSet:
+        assert n.bls_bft.bls_key_register.get_key_by_name(txnPoolNodeSet[0].name) is None
+
+
+def test_fail_ordering_with_nodes_sign_without_key_proof(looper,
+                                                          txnPoolNodeSet,
+                                                          sdk_pool_handle,
+                                                          sdk_wallet_stewards,
+                                                          sdk_wallet_client,
+                                                          monkeypatch):
+    '''
+    Add BLS key without BLS key proof for 3 nodes. Test that when VALIDATE_SIGN_WITHOUT_BLS_KEY_PROOF = False
+    node does not use key sent without proof and transaction can not be ordered.
+    '''
+
+    for n in txnPoolNodeSet:
+        monkeypatch.setattr(n.poolManager.reqHandler, 'doStaticValidation', lambda req: True)
+        n.bls_bft.bls_key_register._pool_manager.config.VALIDATE_SIGN_WITHOUT_BLS_KEY_PROOF = True
+    for node_index in range(0, len(txnPoolNodeSet) - 1):
+        update_bls_keys(node_index, sdk_wallet_stewards, sdk_pool_handle, looper, txnPoolNodeSet)
+    monkeypatch.undo()
+    for n in txnPoolNodeSet:
+        n.bls_bft.bls_key_register._pool_manager.config.VALIDATE_SIGN_WITHOUT_BLS_KEY_PROOF = False
+    with pytest.raises(PoolLedgerTimeoutException):
+        sdk_send_random_and_check(looper, txnPoolNodeSet,
+                                  sdk_pool_handle, sdk_wallet_stewards[3], 1, total_timeout=10)
+
+
+def update_bls_keys(node_index, sdk_wallet_stewards, sdk_pool_handle, looper, txnPoolNodeSet):
```

Comment:
Please rename to `update_bls_keys_no_proof`

Reply:
Done